src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html

```diff
@@ -43,38 +42,29 @@ <h3>{{ 'arTeMiSApp.courseOverview.exerciseDetails.problemStatement' | translate
             <!-- RESULTS START -->
             <div class="row mb-2 align-items-baseline" *ngIf="showResults">
                 <div class="col-auto">
-                    <h3>{{ 'arTeMiSApp.courseOverview.exerciseDetails.yourResult' | translate }}</h3>
+                    <h3>{{ 'arTeMiSApp.courseOverview.exerciseDetails.yourGradedResult' | translate }}</h3>
                 </div>
             </div>
-            <div class="row mb-2" *ngIf="showResults && currentResult">
+            <div class="row mb-2" *ngIf="showResults && currentResult?.rated">
                 <div class="col-12 col-md-12">
                     <div class="row justify-content-between">
                         <div class="col">
                             <h4>
-                                <span class="badge w-100 badge-success" [ngClass]="exerciseRatedBadge(currentResult)">
-                                    {{
-                                        (currentResult.rated ? 'arTeMiSApp.courseOverview.exerciseDetails.graded' : 'arTeMiSApp.courseOverview.exerciseDetails.notGraded')
-                                            | translate
-                                            | uppercase
-                                    }}
+                                <span class="badge w-100 badge-success">
+                                    {{ 'arTeMiSApp.courseOverview.exerciseDetails.graded' | translate | uppercase }}
                                 </span>
                             </h4>
                         </div>
                         <div class="col text-left">
-                            <jhi-result [participation]="currentResult.participation" [result]="currentResult" short="true"></jhi-result>
+                            <jhi-result [participation]="currentResult.participation" [result]="currentResult" showUngradedResults="true" short="true"></jhi-result>
```

Comment:
why dir you add `showUngradedResults="true"`? in line 48 you check for currentResult.rated, so we probably dont need it

Reply:
you are right, thanks. this is a leftover from a previous version, I removed it.